Can not build `iceoryx` with gcc 9.4

## Required information
Building `iceoryx` from source using gcc-9.4 yield the following error 

```
[ 17%] Building CXX object hoofs/test/CMakeFiles/hoofs_moduletests.dir/moduletests/test_cxx_vector.cpp.o
In file included from /home/ziad.mostafa/ade-home/iceoryx/iceoryx_hoofs/container/include/iox/uninitialized_array.hpp:119,
                 from /home/ziad.mostafa/ade-home/iceoryx/iceoryx_hoofs/include/iceoryx_hoofs/cxx/vector.hpp:24,
                 from /home/ziad.mostafa/ade-home/iceoryx/iceoryx_hoofs/test/moduletests/test_cxx_vector.cpp:18:
/home/ziad.mostafa/ade-home/iceoryx/iceoryx_hoofs/container/include/iox/detail/uninitialized_array.inl: In member function ‘virtual void {anonymous}::vector_test_EmplaceAtPositionAfterEndBeforeCapacityExceedsFails_Test::TestBody()’:
/home/ziad.mostafa/ade-home/iceoryx/iceoryx_hoofs/container/include/iox/detail/uninitialized_array.inl:40:72: error: iteration 2305843009213693951 invokes undefined behavior [-Werror=aggressive-loop-optimizations]
   40 |     return *reinterpret_cast<const ElementType*>(&m_buffer.value[index]);
      |                                                                        ^
In file included from /home/ziad.mostafa/ade-home/iceoryx/iceoryx_hoofs/include/iceoryx_hoofs/cxx/vector.hpp:239,
                 from /home/ziad.mostafa/ade-home/iceoryx/iceoryx_hoofs/test/moduletests/test_cxx_vector.cpp:18:
/home/ziad.mostafa/ade-home/iceoryx/iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/vector.inl:188:48: note: within this loop
  188 |     for (uint64_t i{sizeBeforeEmplace - 1U}; i > position; --i)
      |                                              ~~^~~~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [hoofs/test/CMakeFiles/hoofs_moduletests.dir/build.make:466: hoofs/test/CMakeFiles/hoofs_moduletests.dir/moduletests/test_cxx_vector.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1218: hoofs/test/CMakeFiles/hoofs_moduletests.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
```

**Operating system:**
 Ubuntu 20.04 LTS

**Compiler version:**
gcc version 9.4.0 (Ubuntu 9.4.0-1ubuntu1~20.04.1) 

**Observed result or behaviour:**
A clear and precise description of the observed result.

`EmplaceAtPositionAfterEndBeforeCapacityExceedsFails` testcase fails due to `error: iteration 2305843009213693951 invokes undefined behavior [-Werror=aggressive-loop-optimizations]`

**NOTE: a strange behavior, when I swapped line 1515 and 1516, the testcase passed** 

**Expected result or behaviour:**
What do you expect to happen?
iceoryx should be built without errors as the `installation` did not specify a version upper-limit for gcc 
`[GCC](https://gcc.gnu.org/), 8.3 or later (5.4 currently supported too)`

**Conditions where it occurred / Performed steps:**
Describe how one can reproduce the bug.
1. clone `iceoryx` 
2.  `cmake -Bbuild -Hiceoryx_meta -DBUILD_ALL=ON -DBUILD_SHARED_LIBS=ON -DEXAMPLES=ON -DINTROSPECTION=ON -DBUILD_STRICT=ON -DBUILD_TEST=OFF`
3. `cmake --build build`

--- a/doc/website/release-notes/iceoryx-unreleased.md
+++ b/doc/website/release-notes/iceoryx-unreleased.md
@@ -67,6 +67,7 @@
 - `WaitSet::wait` returns if data was send before `WaitSet::attachState(.., State::HAS_{DATA, REQUEST, RESPONSE})` [\#1855](https://github.com/eclipse-iceoryx/iceoryx/issues/1855)
 - Provide a better error message when attempting to create a shared memory in read-only mode
   [\#1821](https://github.com/eclipse-iceoryx/iceoryx/issues/1821)
+- Can not build iceoryx with gcc 9.4 [\#1871](https://github.com/eclipse-iceoryx/iceoryx/issues/1871)
 
 **Refactoring:**
 

--- a/iceoryx_hoofs/test/moduletests/test_cxx_vector.cpp
+++ b/iceoryx_hoofs/test/moduletests/test_cxx_vector.cpp
@@ -1512,7 +1512,9 @@ TEST_F(vector_test, EmplaceAtPositionAfterEndBeforeCapacityExceedsFails)
     sut.emplace_back(0U);
     sut.emplace_back(1U);
 
-    EXPECT_FALSE(sut.emplace(sut.size() + 1, 3U));
-    ASSERT_THAT(sut.size(), Eq(2U));
+    constexpr uint64_t EXPECTED_SIZE{2};
+    ASSERT_THAT(sut.size(), EXPECTED_SIZE);
+    EXPECT_FALSE(sut.emplace(EXPECTED_SIZE + 1, 3U));
+    ASSERT_THAT(sut.size(), EXPECTED_SIZE);
 }
 } // namespace

